src/databricks/labs/dqx/check_funcs.py

```diff
@@ -746,6 +834,97 @@ def is_ipv4_address_in_cidr(column: str | Column, cidr_block: str) -> Column:
 
 
 @register_rule("row")
+def is_valid_ipv6_address(column: str | Column) -> Column:
+    """
+    Checks whether a column contains properly formatted IPv6 addresses.
+
+    This rule checks for four accepted IPv6 formats:
+      - Fully uncompressed (e.g. '2001:0db8:0000:0000:0000:0000:0000:0001')
+      - Compressed (e.g. '2001:db8::1')
+      - Loopback ('::1')
+      - Unspecified ('::')
+
+    A value fails the check if it does not match **any** of these valid IPv6 patterns.
+
+    :param column: column to check; can be a string column name or a column expression
```

Comment:
please convert all docstrings to google style docs. We recently converted all docstrings, as there is better support for api generation from these